### PR TITLE
feat: Added `Expr.any` and `Expr.all` to Dask

### PR DESCRIPTION
**Pull request by mistShard**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue #637 
- Closes #

## Checklist

- [x] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

## If you have comments or can explain your changes, please do so below.
Sorry for taking 3 PR's to resolve merge conflicts.

### CODE REVIEW

**Comment by mistShard** [comment]
@MarcoGorelli hope this works.

File: narwhals/_dask/expr.py [hunk 1]
```diff
@@ -438,6 +438,34 @@ def abs(self) -> Self:
             returns_scalar=False,
         )
 
+    def all(
+        self,
+        *,
+        ignore_nulls: bool = True,
```
Comment (MarcoGorelli) [inline narwhals/_dask/expr.py:444]: we don't pass down any arguments in Narwhals, I think you can just remove it from here and set the value which matches Polars' default in the function body

**Comment by mistShard** [comment]
@MarcoGorelli does this work?

**Review by MarcoGorelli — APPROVED** [review]
amazing, thanks @mistShard !

**Pull request merged**